File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] res = new int[n];
        Stack<Integer> stk = new Stack<>();
        
        for (int i = 0; i < n; i++) {
            while (!stk.isEmpty() && temperatures[stk.get(stk.size()-1)] < temperatures[i]) {
                int idx = stk.pop();
                res[idx] = i - idx;
            }
            stk.push(i);
        }
        return res;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        stk = []
        res = [0]*n
        for i in range(n):
            while stk and temperatures[stk[-1]] < temperatures[i]:
                val = stk.pop()
                res[val] = i - val
            stk.append(i)

        return res